integration-test/src/main/java/com/sequenceiq/it/cloudbreak/listener/ReportListener.java

```diff
@@ -24,23 +23,31 @@
 
     public static final String MEASUREMENTS = "measurements";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(ReportListener.class);
-
     @Override
     public void onTestFailure(ITestResult tr) {
-        super.onTestFailure(tr);
         logUrl(tr);
         logMeasurements(tr);
         log(tr);
+        Reporter.setCurrentTestResult(tr);
+        super.onTestFailure(tr);
     }
 
     @Override
     public void onTestSuccess(ITestResult tr) {
         logUrl(tr);
         logMeasurements(tr);
+        Reporter.setCurrentTestResult(tr);
```

Comment:
Just asking, would this work correctly with parallel runs since it's static?

Reply:
Just checked in the code, it uses ThreadLocal internally.

Reply:
Each test method and its respective before and after method is executed in a different thread. This is identified by the ID of the thread that is printed on the console.
You can find some additional details, for example: [TestNG parallel execution of tests, classes and suites](https://howtodoinjava.com/testng/testng-executing-parallel-tests/)

Reply:
You were scratcing the ugly truth: After I've tested this out with several threads, I had to realised the original solution wasn't able to get the correct throwable to the defined test case. So I've applied the same prefixing as we already have for `searchUrl`.
So finally not the `Reporter` caused the issue as rather than `custom.results.xml` template.

Reply:
Here is the test upload for Quanta: [https://quanta.infra.cloudera.com/?#/runDetails?gtn=6369877](https://quanta.infra.cloudera.com/?#/runDetails?gtn=6369877)